docker-plugin/src/main/java/io/micronaut/gradle/docker/NativeImageDockerfile.java

```diff
@@ -462,11 +464,8 @@ private CopyFileInstruction toCopyResourceDirectoryInstruction(java.io.File reso
             options.getBuildArgs().add("--report-unsupported-elements-at-runtime");
         } else if (buildStrategy == DockerBuildStrategy.LAMBDA) {
             JavaApplication javaApplication = getProject().getExtensions().getByType(JavaApplication.class);
-            if (!javaApplication.getMainClass().isPresent()) {
-                options.getMainClass().set("io.micronaut.function.aws.runtime.MicronautLambdaRuntime");
-            }
-            if (!options.getMainClass().isPresent()) {
```

Comment:
@melix should not this be if/else if instead of or. Maybe this is the cause of the issue we saw.

Reply:
No it shouldn't. But I found some plugins which override each other's defaults, which is the real issue. This thing is spaghetti code...